cmd/podman/patch.go

```
@@ -0,0 +1,203 @@
+    }
+)
+
+func init() {
+        patchCommand.Command = _patchCommand
+        flags := patchCommand.Flags()
+        flags.BoolVarP(
+            &patchCommand.All,
+            "all",
+            "a",
+            false,
+            "Apply the patch to all the running containers")
+        flags.BoolVarP(
+            &patchCommand.Ignore,
+            "ignore-fail",
+            "i",
+            false,
+            "Ignore if the file to patch doesn't exist in a container")
+
+        rootCmd.AddCommand(patchCommand.Command)
+}
+
+// Define the entrypoint command
+func patchCmd(c *cliconfig.PatchValues) error {
+//func patchCmd(c *cli.Context) error {
+    runtime, err := libpodruntime.GetRuntime(&c.PodmanCommand)
+    //runtime, err := libpodruntime.GetRuntime(c)
+    if os.Geteuid() != 0 {
+        rootless.SetSkipStorageSetup(true)
+    }
+
+    if os.Geteuid() != 0 {
+        if driver := runtime.GetConfig().StorageConfig.GraphDriverName; driver != "vfs" {
+            // Do not allow to mount a graphdriver that is not vfs if we are
+            // creating the userns as part
+            // of the mount command.
+            fmt.Errorf("cannot mount using driver %s in rootless mode", driver)
+        }
+
+        became, ret, err := rootless.BecomeRootInUserNS()
+        if err != nil {
+            return err
+        }
+        if became {
+            os.Exit(ret)
+        }
+    }
+    args := c.InputArgs
+    if len(args) < 3 {
+        if c.All && len(args) < 2 {
+            return errors.Errorf("too few given arguments")
+        }
+    }
+
+    if err != nil {
+        return errors.Wrapf(err, "could not get runtime")
+    }
+    defer runtime.Shutdown(false)
+
+    dst := args[0]
+    patch := args[1]
+
+    var containers []string
+    if ! c.All {
+        containers = args[2:]
+    } else {
+        containers = retrieveExisting(runtime)
+    }
+
+    if ! canPatch() {
+        fmt.Println("Patch command not on your system")
```

I think we prefer to return an error here, instead of manually printing and invoking os.Exit